plugins/mac-avcapture/plugin-main.m

```diff
@@ -79,42 +79,52 @@ static void av_fast_capture_set_defaults(obs_data_t *settings)
 {
     OBSAVCapture *capture = (__bridge OBSAVCapture *) (av_capture);
     OBSAVCaptureInfo *capture_info = capture.captureInfo;
+    AVCaptureDevice *device = capture.deviceInput.device;
+    NSString *effectsWarningKey = [OBSAVCapture effectsWarningForDevice:device];
+    bool has_effects = effectsWarningKey != nil;
+    const char *effects_warning_string;
+    if (has_effects) {
+        effects_warning_string = obs_module_text(effectsWarningKey.UTF8String);
+    }
 
     obs_properties_t *properties = obs_properties_create();
 
     // Create Properties
     obs_property_t *device_list = obs_properties_add_list(properties, "device", obs_module_text("Device"),
                                                           OBS_COMBO_TYPE_LIST, OBS_COMBO_FORMAT_STRING);
+
+    obs_property_t *effects_warning;
+    if (has_effects) {
+        effects_warning = obs_properties_add_text(properties, "effects_warning", effects_warning_string, OBS_TEXT_INFO);
```

Comment:
Why not use the return of `obs_module_text(effectsWarningKey.UTF8String)` here directly and also use `if (effectsWarningKey)` to check for a valid key (`effectsWarningForDevice` returns nil if no warnings are available).

This avoids an additional check for `has_effects` and the declaration of a boolean and a `char` array. `effectsWarningKey` can be used for the later check as well.

Reply:
Thanks, fixed.